### PR TITLE
docs: document how osmet works

### PR DESCRIPTION
**Pull request by jlebon**
This was requested a few times.

Closes: #463

### CODE REVIEW

File: docs/osmet.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+It then "packs" the raw image by going through it but skipping all the
+chunks which correspond to mapped OSTree objects. The resulting packed
+image then essentially only contains data like partition tables, the
+BIOS boot partition, inode metadata, etc...
+
+This packed image is passed through an xz filter and then bundled
+together with the serialized OSTree object table into an "osmet" file
+which is included as part of the rootfs initrd in the live ISO and PXE
+environments.
```
Comment (bgilbert) [inline docs/osmet.md:40]: Optional: say more explicitly that the osmet files are not in the squashfs.

File: docs/osmet.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+At install time (i.e. when users boot the live environment),
+coreos-installer detects the osmet files present and uses it to recreate
```
Comment (bgilbert) [inline docs/osmet.md:43]: Would be good to say that there are separate osmet files for 512b and 4K sector disks, and coreos-installer automatically picks one.

File: docs/osmet.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+parent: Development
+nav_order: 2
+---
+
+# What is osmet?
+
+CoreOS systems support booting into live PXE and ISO environments. These
+work using a "rootfs initrd" which contains a squashfs of the actual
+rootfs to boot into. This rootfs, like any OSTree-based system, contains
+the system OSTree repo from which objects a hardlinked out to populate
```
Comment (bgilbert) [inline docs/osmet.md:12]: `objects are`

File: docs/osmet.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+it.
```
Comment (bgilbert) [inline docs/osmet.md:13]: s/it/the root filesystem tree/?

File: docs/osmet.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+environments.
+
+At install time (i.e. when users boot the live environment),
+coreos-installer detects the osmet files present and uses it to recreate
+the metal image to write to disk. The unpacking process is the inverse
+of packing: it decompresses through xz, then with the deserialized
+lookup table, it uses the OSTree objects from the mounted squashfs to
+fill in the gaps which the packed object skipped over. Simultaneously,
+it verifies the checksum of the written image to ensure we still have
+bit-for-bit matching.
```
Comment (bgilbert) [inline docs/osmet.md:49]: Slightly clearer: `to ensure that it exactly matches the original`.

**Review by bgilbert — COMMENTED** [review]
Thanks for writing this!

**Comment by jlebon** [comment]
Fixes!

File: docs/osmet.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+At install time (i.e. when users boot the live environment),
+coreos-installer detects the osmet files present and uses the
+appropriate one for the sector size of the target disk to recreate the
+metal image to write to disk. The unpacking process is the inverse of
+packing: it decompresses through xz, then with the deserialized lookup
+table, it uses the OSTree objects from the mounted squashfs to fill in
+the gaps which the packed object skipped over. Simultaneously, it
+verifies the checksum of the written image to ensure that it matches
+exactly the original.
```
Comment (bgilbert) [inline docs/osmet.md:54]: `exactly matches` would be more idiomatic.

**Review by bgilbert — APPROVED** [review]
:tada:

File: docs/osmet.md (hunk 6 above)
Comment (jlebon) [reply docs/osmet.md:54]: Done!

**Review by bgilbert — APPROVED** [review]

**Pull request merged**